src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -96,24 +96,10 @@ struct LegacyDefaultCallback
                                   PairValueIndex<Value, Index> const &value,
                                   OutputFunctor const &output) const
   {
-    output(value.index);
+    output((int)value.index);
   }
 };
 
-#ifdef ARBORX_ENABLE_MPI
-struct LegacyDefaultCallbackWithRank
```

Comment:
I moved this to distributed headers and renamed. It no longer is being called only from APIv1, but also from within APIv2.